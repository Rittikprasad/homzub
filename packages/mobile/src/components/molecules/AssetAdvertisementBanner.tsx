import React from 'react';
import { StyleSheet, View } from 'react-native';
import { DashboardRepository } from '@homzhub/common/src/domain/repositories/DashboardRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
import { PaginationComponent } from '@homzhub/mobile/src/components/atoms/PaginationComponent';
import { SVGUri } from '@homzhub/common/src/components/atoms/Svg';
import { AssetAdvertisement, AssetAdvertisementResults } from '@homzhub/common/src/domain/models/AssetAdvertisement';

interface IAssetAdvertisementBannerState {
  banners: AssetAdvertisement;
  activeSlide: number;
}

const ITEM_WIDTH = theme.viewport.width - theme.layout.screenPadding * 2;
export class AssetAdvertisementBanner extends React.PureComponent<{}, IAssetAdvertisementBannerState> {
  public state = {
    banners: {} as AssetAdvertisement,
    activeSlide: 0,
  };

  public componentDidMount = async (): Promise<void> => {
    await this.fetchAdvertisementBanners();
  };

  public render(): React.ReactNode {
    const { banners, activeSlide } = this.state;

    if (banners?.results?.length <= 0 || !banners) {
      return null;
    }

    return (
      <View style={styles.container}>
        <SnapCarousel
          carouselData={banners?.results ?? []}
          carouselItem={this.renderCarouselItem}
          sliderWidth={ITEM_WIDTH}
          itemWidth={ITEM_WIDTH}
          activeIndex={activeSlide}
          containerStyle={styles.radius}
          onSnapToItem={this.onSnapToItem}
          testID="bannerSnap"
        />
        <PaginationComponent
          dotsLength={banners?.results?.length ?? 0}
          activeSlide={activeSlide}
          containerStyle={styles.overlay}
          activeDotStyle={styles.activeDotStyle}
          inactiveDotStyle={styles.inactiveDotStyle}
        />
      </View>
    );
  }

  private renderCarouselItem = (item: AssetAdvertisementResults): React.ReactElement => {
    return <SVGUri uri={item.attachment.link} />;
  };

  public onSnapToItem = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };

  public fetchAdvertisementBanners = async (): Promise<void> => {
    const response: AssetAdvertisement = await DashboardRepository.getAdvertisements({ category: 'general' });
    this.setState({ banners: response });
  };
}

const styles = StyleSheet.create({
  container: {
    marginTop: 16,
    height: 200,
    backgroundColor: theme.colors.white,
  },
  radius: {
    borderRadius: 4,
  },
  overlay: {
    position: 'absolute',
    alignSelf: 'center',
    bottom: 0,
    paddingVertical: 10,
  },
  activeDotStyle: {
    borderWidth: 1,
  },
  inactiveDotStyle: {
    backgroundColor: theme.colors.darkTint6,
    borderColor: theme.colors.transparent,
  },
});
