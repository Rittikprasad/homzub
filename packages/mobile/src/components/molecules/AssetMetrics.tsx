import React from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, ViewStyle, View } from 'react-native';
import { useSelector } from 'react-redux';
import { theme } from '@homzhub/common/src/styles/theme';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { PricePerUnit } from '@homzhub/common/src/components/atoms/PricePerUnit';

interface IProps {
  header: string;
  colorCode: string;
  selectedAssetType?: string;
  value: string | number;
  isCurrency: boolean;
  cardStyle?: StyleProp<ViewStyle>;
  testID?: string;
  onPressMetrics?: () => void;
  maxLength?: number;
}

const AssetMetrics = (props: IProps): React.ReactElement => {
  const { header, value, cardStyle, testID, onPressMetrics, isCurrency, selectedAssetType, colorCode, maxLength } =
    props;

  const currency = useSelector(UserSelector.getCurrency);

  const handlePress = (): void => {
    if (onPressMetrics) {
      onPressMetrics();
    }
  };

  const renderItem = (): React.ReactElement => {
    return (
      <>
        <Text type="small" textType="semiBold" style={styles.metricName} maxLength={maxLength}>
          {header}
        </Text>
        {isCurrency ? (
          <PricePerUnit
            textStyle={styles.metricName}
            currency={currency}
            priceTransformation={false}
            price={value as number}
          />
        ) : (
          <Text type="large" textType="semiBold" style={styles.metricName}>
            {value}
          </Text>
        )}
      </>
    );
  };

  return (
    <TouchableOpacity onPress={handlePress} testID={testID}>
      <View
        style={[
          styles.container,
          cardStyle,
          { backgroundColor: colorCode },
          header === selectedAssetType && styles.selectedItem,
        ]}
      >
        {renderItem()}
      </View>
    </TouchableOpacity>
  );
};

const memoizedComponent = React.memo(AssetMetrics);
export { memoizedComponent as AssetMetrics };

const styles = StyleSheet.create({
  container: {
    borderRadius: 8,
    paddingVertical: 8,
    marginVertical: 12,
    marginStart: 12,
  },
  metricName: {
    textAlign: 'center',
    marginVertical: 4,
    color: theme.colors.white,
  },
  selectedItem: {
    borderWidth: 1.25,
    borderColor: theme.colors.white,
    shadowColor: theme.colors.shadow,
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.32,
    shadowRadius: 5.46,
    elevation: 9,
  },
});
