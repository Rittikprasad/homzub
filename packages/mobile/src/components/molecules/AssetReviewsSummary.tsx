import React, { useCallback, useState } from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import { SnapCarousel } from '@homzhub/mobile/src/components/atoms/Carousel';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { Label, Title } from '@homzhub/common/src/components/atoms/Text';
import { PaginationComponent } from '@homzhub/mobile/src/components/atoms/PaginationComponent';
import { Rating } from '@homzhub/common/src/components/atoms/Rating';
import { AssetReview } from '@homzhub/common/src/domain/models/AssetReview';
import { Pillar } from '@homzhub/common/src/domain/models/Pillar';

interface IProps {
  reviewSummary: AssetReview;
  showDivider?: boolean;
  titleRequired?: boolean;
  containerStyle?: StyleProp<ViewStyle>;
  sliderWidth?: number;
}

const SLIDER_WIDTH = theme.viewport.width - theme.layout.screenPadding * 4;

const AssetReviewsSummary = ({
  reviewSummary,
  containerStyle,
  titleRequired = true,
  showDivider = true,
  sliderWidth = SLIDER_WIDTH,
}: IProps): React.ReactElement => {
  const { t } = useTranslation();
  const [activeIndex, setActiveIndex] = useState(0);
  const { reviewCount, rating, pillarRatings } = reviewSummary;

  const splitData = useCallback((): Pillar[][] => {
    const newArr = [];

    for (let i = 0; i < pillarRatings.length; i += 3) {
      newArr.push(pillarRatings.slice(i, i + 3));
    }

    return newArr;
  }, [pillarRatings]);

  const renderItem = (item: Pillar[]): React.ReactElement => {
    return (
      <View style={styles.ratingContainer}>
        {item.map(
          (rat: Pillar): React.ReactElement => (
            <Rating circle key={rat.pillarName?.id} value={rat.rating} title={rat.pillarName?.name} />
          )
        )}
      </View>
    );
  };

  return (
    <>
      <View style={containerStyle}>
        {titleRequired && (
          <Label type="large" textType="semiBold" style={styles.headerTitle}>
            {t('common:overall')}
          </Label>
        )}
        <View style={styles.overallContainer}>
          <Title type="small">{rating}</Title>
          {rating && <Rating value={rating} />}
          <Label style={styles.basedOn}>{t('property:basedOn', { count: reviewCount })}</Label>
        </View>
        <SnapCarousel
          carouselData={splitData()}
          carouselItem={renderItem}
          activeIndex={activeIndex}
          onSnapToItem={setActiveIndex}
          sliderWidth={sliderWidth}
          itemWidth={sliderWidth}
        />
        <PaginationComponent
          dotsLength={splitData().length}
          activeSlide={activeIndex}
          containerStyle={styles.paginationContainer}
          activeDotStyle={[styles.dot, styles.activeDot]}
          inactiveDotStyle={[styles.dot, styles.inactiveDot]}
        />
      </View>
      {showDivider && <Divider containerStyles={styles.divider} />}
    </>
  );
};

const styles = StyleSheet.create({
  ratingContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  overallContainer: {
    marginTop: 16,
    marginBottom: 24,
    alignItems: 'center',
  },
  paginationContainer: {
    paddingVertical: 16,
  },
  dot: {
    width: 7,
    height: 7,
  },
  activeDot: {
    borderWidth: 1.5,
  },
  inactiveDot: {
    backgroundColor: theme.colors.disabled,
    borderWidth: 0,
  },
  headerTitle: {
    marginTop: 16,
    color: theme.colors.darkTint3,
  },
  basedOn: {
    color: theme.colors.darkTint4,
    marginTop: 4,
  },
  divider: {
    borderColor: theme.colors.background,
  },
});

const memoizedComponent = React.memo(AssetReviewsSummary);
export { memoizedComponent as AssetReviewsSummary };
