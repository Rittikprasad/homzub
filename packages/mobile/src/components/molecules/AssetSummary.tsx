import React, { useState, useEffect } from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, ViewStyle } from 'react-native';
import { useTranslation } from 'react-i18next';
import LinearGradient from 'react-native-linear-gradient';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IProps {
  notification?: number;
  serviceTickets?: number;
  dues?: number;
  messages?: number;
  jobs?: number;
  isFFM?: boolean;
  onPressNotification?: () => void;
  onPressServiceTickets?: () => void;
  onPressDue?: () => void;
  onPressMessages?: () => void;
  containerStyle?: StyleProp<ViewStyle>;
}

interface IItem {
  icon: string;
  color: string;
  title: string;
  onPress?: () => void;
  count: number;
}

const AssetSummary = (props: IProps): React.ReactElement => {
  const { t } = useTranslation(LocaleConstants.namespacesKey.assetDashboard);
  const {
    notification = 0,
    serviceTickets = 0,
    dues = 0,
    messages = 0,
    jobs = 0,
    isFFM = false,
    containerStyle,
    onPressDue,
    onPressNotification,
    onPressServiceTickets,
    onPressMessages,
  } = props;

  // Data for the static part of the component
  const [data, setData] = useState<IItem[]>([]);
  useEffect(() => {
    setData([
      {
        icon: icons.alert,
        color: theme.colors.notificationGreen,
        title: t('notification'),
        onPress: onPressNotification,
        count: notification,
      },
      {
        icon: icons.serviceRequest,
        color: theme.colors.orange,
        title: t('assetMore:tickets'),
        onPress: onPressServiceTickets,
        count: serviceTickets,
      },
      {
        icon: icons.chat,
        color: isFFM ? theme.colors.danger : theme.colors.blue,
        title: isFFM ? t('Jobs') : t('moreSettings:messagesText'),
        onPress: onPressMessages,
        count: isFFM ? jobs : messages,
      },
    ]);
  }, [
    dues,
    notification,
    messages,
    serviceTickets,
    onPressDue,
    onPressNotification,
    onPressServiceTickets,
    onPressMessages,
    t,
  ]);

  return (
    <LinearGradient
      useAngle
      angle={180}
      colors={[theme.colors.white, theme.colors.background]}
      locations={[0, 1]}
      style={[styles.container, containerStyle]}
    >
      {data.map((item, index) => {
        const { icon, title, color, count, onPress } = item;
        return (
          <TouchableOpacity
            key={title}
            style={[styles.item, index !== data.length - 1 && styles.divider]}
            onPress={onPress}
          >
            <Icon name={icon} color={color} size={25} />
            <Text type="small" textType="semiBold" style={{ color }} numberOfLines={1}>
              {title}
            </Text>
            <Text type="regular" textType="bold" style={{ color }}>
              {count}
            </Text>
          </TouchableOpacity>
        );
      })}
    </LinearGradient>
  );
};

export { AssetSummary };

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    borderRadius: 4,
  },
  item: {
    flex: 0.33,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 12,
  },
  divider: {
    borderEndWidth: 1.5,
    borderEndColor: theme.colors.disabled,
  },
});
