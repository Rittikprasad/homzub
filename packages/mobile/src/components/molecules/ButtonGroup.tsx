import React from 'react';
import { TouchableOpacity, StyleSheet, View, ViewStyle, StyleProp } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { Label } from '@homzhub/common/src/components/atoms/Text';

interface IButtonGroupItem<T> {
  title: string;
  value: T;
}

interface IButtonGroupProps<T> {
  data: IButtonGroupItem<T>[];
  onItemSelect: (value: T) => void;
  selectedItem: T;
  containerStyle?: StyleProp<ViewStyle>;
  buttonItemStyle?: StyleProp<ViewStyle>;
  testID?: string;
}

export class ButtonGroup<T> extends React.PureComponent<IButtonGroupProps<T>> {
  public render(): React.ReactElement {
    const { data, containerStyle = {} } = this.props;
    return (
      <View style={[styles.container, containerStyle]}>
        {data.map((item: IButtonGroupItem<T>, index) => {
          return this.renderItem(item, index);
        })}
      </View>
    );
  }

  public renderItem = (item: IButtonGroupItem<T>, index: number): React.ReactElement => {
    const { selectedItem, onItemSelect, buttonItemStyle = {}, testID = 'btngrp' } = this.props;

    // Styles
    const isSelected = item.value === selectedItem;
    const textStyle = StyleSheet.flatten([styles.textStyle, isSelected && styles.selectedTextStyle]);
    const buttonItemContainerStyle = StyleSheet.flatten([
      styles.item,
      isSelected && styles.selectedContainerStyle,
      index % 2 === 0 && { marginEnd: 16 },
      buttonItemStyle,
    ]);

    const onItemPress = (): void => onItemSelect(item.value);

    return (
      <TouchableOpacity onPress={onItemPress} style={buttonItemContainerStyle} key={item.title} testID={testID}>
        <Label type="large" textType={isSelected ? 'semiBold' : 'regular'} style={textStyle}>
          {item.title}
        </Label>
      </TouchableOpacity>
    );
  };
}

const styles = StyleSheet.create({
  container: {
    flexWrap: 'wrap',
    flexDirection: 'row',
  },
  item: {
    flex: 0.5,
    paddingVertical: 6,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 4,
    borderWidth: 1,
    borderColor: theme.colors.disabled,
    backgroundColor: theme.colors.white,
  },
  selectedContainerStyle: {
    backgroundColor: theme.colors.primaryColor,
  },
  textStyle: {
    color: theme.colors.darkTint5,
  },
  selectedTextStyle: {
    color: theme.colors.white,
  },
});
