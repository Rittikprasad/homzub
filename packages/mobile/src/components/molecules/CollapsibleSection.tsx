import React, { useState } from 'react';
import { TouchableOpacity, StyleSheet, View, StyleProp, ViewStyle, TextStyle } from 'react-native';
import Collapsible from 'react-native-collapsible';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { Text } from '@homzhub/common/src/components/atoms/Text';

interface ICollapsibleSectionProps {
  title: string;
  icon?: string;
  children: React.ReactNode;
  onCollapse?: (isCollapsed: boolean) => void;
  isDividerRequired?: boolean;
  titleStyle?: StyleProp<TextStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  initialCollapsedValue?: boolean;
  isCollapsibleRequired?: boolean;
  collapseIcon?: string;
  expandIcon?: string;
  iconStyle?: StyleProp<ViewStyle>;
}
const CollapsibleSection = (props: ICollapsibleSectionProps): React.ReactElement => {
  const {
    title,
    children,
    initialCollapsedValue = false,
    icon,
    titleStyle,
    isDividerRequired = false,
    onCollapse,
    isCollapsibleRequired = true,
    containerStyle,
    collapseIcon = icons.minus,
    expandIcon = icons.plus,
    iconStyle,
  } = props;
  const [isCollapsed, setIsCollapsed] = useState(initialCollapsedValue);

  const onPress = (): void => {
    setIsCollapsed(!isCollapsed);
    if (onCollapse) {
      onCollapse(!isCollapsed);
    }
  };

  return (
    <View style={[styles.container, containerStyle]}>
      <TouchableOpacity
        style={styles.ratingsHeading}
        disabled={!isCollapsibleRequired}
        onPress={onPress}
        testID="collapse"
      >
        <View style={styles.leftView}>
          {icon && <Icon name={icon} size={22} color={theme.colors.darkTint4} />}
          <Text type="small" textType="semiBold" style={[styles.textColor, titleStyle]}>
            {title}
          </Text>
        </View>
        {isCollapsibleRequired && (
          <Icon
            name={isCollapsed ? expandIcon : collapseIcon}
            size={20}
            color={theme.colors.darkTint4}
            style={iconStyle}
          />
        )}
      </TouchableOpacity>
      <Collapsible collapsed={isCollapsed}>{children}</Collapsible>
      {isDividerRequired && <Divider containerStyles={styles.divider} />}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 24,
  },
  divider: {
    marginTop: 24,
    borderColor: theme.colors.darkTint10,
  },
  leftView: {
    flexDirection: 'row',
    alignItems: 'center',
    maxWidth: 250,
  },
  textColor: {
    color: theme.colors.darkTint4,
  },
  ratingsHeading: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});

const memoizedComponent = React.memo(CollapsibleSection);
export { memoizedComponent as CollapsibleSection };
