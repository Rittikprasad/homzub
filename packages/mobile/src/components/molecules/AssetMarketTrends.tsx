import React from 'react';
import { FlatList, ImageBackground, StyleSheet, TouchableOpacity, View, Keyboard } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { debounce } from 'lodash';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { CommonRepository } from '@homzhub/common/src/domain/repositories/CommonRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { ImagePlaceholder } from '@homzhub/common/src/components/atoms/ImagePlaceholder';
import { Label, Text } from '@homzhub/common/src/components/atoms/Text';
import { EmptyState } from '@homzhub/common/src/components/atoms/EmptyState';
import { SelectionPicker } from '@homzhub/common/src/components/atoms/SelectionPicker';
import { SearchBar } from '@homzhub/common/src/components/molecules/SearchBar';
import { MarketTrendsResults, MarketTrendType } from '@homzhub/common/src/domain/models/MarketTrends';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

interface IProps extends WithTranslation {
  isDashboard?: boolean;
  onViewAll?: () => void;
  onTrendPress: (url: string, id: number) => void;
}

interface IMarketTrendsState {
  data: MarketTrendsResults[];
  selectedType: MarketTrendType;
  searchQuery: string;
}

export class AssetMarketTrends extends React.PureComponent<IProps, IMarketTrendsState> {
  public state = {
    data: [],
    selectedType: MarketTrendType.PERSPECTIVE,
    searchQuery: '',
  };

  private debouncedSearch = debounce(() => {
    this.getMarketTrends(true).then();
  }, 500);

  public componentDidMount = async (): Promise<void> => {
    await this.getMarketTrends(true);
  };

  public render(): React.ReactNode {
    const { t, isDashboard = false } = this.props;
    const { data } = this.state;

    return (
      <View style={[styles.container, isDashboard && styles.headerView]}>
        {this.renderHeader()}
        {data.length > 0 ? (
          <FlatList
            data={data}
            initialNumToRender={30}
            keyExtractor={this.keyExtractor}
            renderItem={this.renderTrends}
            onEndReached={this.onEndReached}
            onEndReachedThreshold={0.8}
            showsVerticalScrollIndicator={false}
          />
        ) : (
          <EmptyState title={t('assetMore:noTrends')} />
        )}
      </View>
    );
  }

  private renderHeader = (): React.ReactNode => {
    const { onViewAll, t, isDashboard = false } = this.props;
    const { data, selectedType, searchQuery } = this.state;

    if (isDashboard) {
      return (
        <View style={styles.header}>
          <Icon name={icons.increase} color={theme.colors.darkTint4} size={20} />
          <Text type="small" textType="semiBold" style={styles.headerText}>
            {t('marketTrends')}
          </Text>
          {data.length > 0 && (
            <Label type="large" textType="semiBold" onPress={onViewAll} style={styles.viewAll}>
              {t('viewAll')}
            </Label>
          )}
        </View>
      );
    }

    return (
      <View>
        <SelectionPicker<MarketTrendType>
          data={[
            { title: 'Perspective', value: MarketTrendType.PERSPECTIVE },
            { title: 'Video', value: MarketTrendType.VIDEO },
          ]}
          selectedItem={[selectedType]}
          onValueChange={this.onTabChange}
        />
        <SearchBar
          placeholder={t('searchByKeyword')}
          value={searchQuery}
          updateValue={this.onUpdateSearchText}
          containerStyle={styles.searchBar}
        />
      </View>
    );
  };

  public renderTrends = ({ item }: { item: MarketTrendsResults }): React.ReactElement => {
    const { title, postedAtDate, link, imageUrl, id, trendType } = item;
    const { onTrendPress } = this.props;

    const onLinkPress = (): void => {
      onTrendPress(link, id);
    };

    return (
      <TouchableOpacity onPress={onLinkPress} style={styles.trendContainer} testID="linkTouch">
        {imageUrl ? (
          <ImageBackground source={{ uri: imageUrl }} style={styles.image}>
            {trendType === MarketTrendType.VIDEO && (
              <View style={styles.videoOverlay}>
                <Icon name={icons.play} size={28} color={theme.colors.white} />
              </View>
            )}
          </ImageBackground>
        ) : (
          <ImagePlaceholder height={80} width={80} containerStyle={styles.placeHolderImage} />
        )}
        <View style={styles.trendValuesContainer}>
          <Label type="large" textType="regular" style={styles.trendHeader} numberOfLines={2}>
            {title}
          </Label>
          <View style={styles.trendData}>
            <Label type="regular" textType="regular" style={styles.trendDate}>
              {postedAtDate}
            </Label>
            <Icon name={icons.dart} color={theme.colors.primaryColor} size={18} />
          </View>
        </View>
      </TouchableOpacity>
    );
  };

  private onTabChange = (newType: MarketTrendType): void => {
    Keyboard.dismiss();
    this.setState({ selectedType: newType, searchQuery: '' }, () => {
      this.getMarketTrends(true).then();
    });
  };

  private onUpdateSearchText = (searchQuery: string): void => {
    this.setState({ searchQuery }, this.debouncedSearch);
  };

  private onEndReached = (): void => {
    const { isDashboard = false } = this.props;
    if (isDashboard) {
      return;
    }
    this.getMarketTrends().then();
  };

  public getMarketTrends = async (reset = false): Promise<void> => {
    const { isDashboard } = this.props;
    const { selectedType, searchQuery, data } = this.state;

    let limit = 10;
    let trend_type: MarketTrendType | undefined = selectedType;
    let q: string | undefined = searchQuery;
    if (isDashboard) {
      limit = 2;
      trend_type = undefined;
      q = undefined;
    }

    try {
      const response = await CommonRepository.getMarketTrends({
        limit,
        trend_type,
        q,
        offset: reset ? 0 : data.length,
      });

      this.setState({ data: reset ? response.results : [...data, ...response.results] });
    } catch (err) {
      AlertHelper.error({ message: ErrorUtils.getErrorMessage(err.details) });
    }
  };

  private keyExtractor = (item: MarketTrendsResults): string => `${item.id}`;
}

export default withTranslation(LocaleConstants.namespacesKey.assetDashboard)(AssetMarketTrends);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    paddingBottom: 0,
    backgroundColor: theme.colors.white,
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 16,
  },
  headerText: {
    marginStart: 12,
    flex: 1,
  },
  trendContainer: {
    flexDirection: 'row',
    marginBottom: 16,
    padding: 12,
    borderRadius: 4,
    borderWidth: 1,
    borderColor: theme.colors.background,
  },
  viewAll: {
    color: theme.colors.primaryColor,
  },
  trendDate: {
    color: theme.colors.darkTint5,
  },
  trendValuesContainer: {
    flex: 1,
    justifyContent: 'space-between',
    marginStart: 12,
  },
  trendData: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  trendHeader: {
    color: theme.colors.darkTint2,
    flexWrap: 'wrap',
  },
  videoOverlay: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.colors.overlay,
  },
  image: {
    width: 80,
    height: 80,
    borderRadius: 4,
  },
  placeHolderImage: {
    borderColor: theme.colors.darkTint10,
    borderWidth: 1,
    borderRadius: 4,
  },
  headerView: {
    marginTop: 16,
  },
  searchBar: {
    marginVertical: 16,
  },
});
