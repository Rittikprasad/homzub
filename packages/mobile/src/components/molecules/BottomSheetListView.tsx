import React, { Component } from 'react';
import { FlatList, PickerItemProps, StyleSheet, StyleProp, ViewStyle, TextStyle } from 'react-native';
import { theme } from '@homzhub/common/src/styles/theme';
import { Divider } from '@homzhub/common/src/components/atoms/Divider';
import { ListItem } from '@homzhub/common/src/components/atoms/ListItem';
import { BottomSheet } from '@homzhub/common/src/components/molecules/BottomSheet';

export interface IListItem extends PickerItemProps {
  isNegative?: boolean;
  isDisable?: boolean;
}
interface IProps<T> {
  data: IListItem[];
  selectedValue: string;
  listTitle: string;
  listHeight?: number;
  isBottomSheetVisible: boolean;
  onCloseDropDown: () => void;
  onSelectItem: (value: T) => void;
  testID?: string;
  numColumns?: number;
  showDivider?: boolean;
  hasFullySpannedItems?: boolean;
  extraContent?: () => React.ReactElement;
}

const LIST_HEIGHT = theme.viewport.height <= theme.DeviceDimensions.SMALL.height ? 450 : 700;
export class BottomSheetListView<T> extends Component<IProps<T>> {
  public render(): React.ReactNode {
    const {
      isBottomSheetVisible,
      onCloseDropDown,
      selectedValue,
      listTitle,
      data,
      listHeight = LIST_HEIGHT,
      numColumns = 1,
      extraContent,
    } = this.props;
    return (
      <BottomSheet
        isShadowView
        sheetHeight={listHeight}
        headerTitle={listTitle}
        visible={isBottomSheetVisible}
        onCloseSheet={onCloseDropDown}
      >
        <>
          {extraContent && extraContent()}
          <FlatList
            data={data}
            renderItem={this.renderSheetItem}
            keyExtractor={this.renderKeyExtractor}
            extraData={selectedValue}
            numColumns={numColumns}
            ItemSeparatorComponent={this.itemSeparator}
          />
        </>
      </BottomSheet>
    );
  }

  private renderSheetItem = ({ item, index }: { item: IListItem; index: number }): React.ReactElement => {
    const { selectedValue, onSelectItem, testID, numColumns = 1, hasFullySpannedItems } = this.props;
    // @ts-ignore
    const onItemSelect = (): void => onSelectItem(item.value);
    const isCheck: boolean = selectedValue === item.value;

    const returnStyles = (): StyleProp<ViewStyle | TextStyle> => {
      if (!hasFullySpannedItems) {
        return [styles.item, numColumns > 1 && styles.itemWidth];
      }
      return undefined;
    };

    return (
      <ListItem
        // @ts-ignore
        listItem={item}
        isCheck={isCheck}
        onItemSelect={onItemSelect}
        key={index}
        testID={testID}
        isDisable={item.isDisable}
        // @ts-ignore
        listItemViewStyle={returnStyles()}
        itemStyle={[item.isNegative && styles.redText, item.isDisable && styles.disable]}
        hasFullySpannedItems={hasFullySpannedItems}
      />
    );
  };

  private renderKeyExtractor = (item: PickerItemProps, index: number): string => {
    const { value } = item;
    return `${value}-${index}`;
  };

  private itemSeparator = (): React.ReactElement | null => {
    const { numColumns, showDivider = true } = this.props;
    if ((numColumns && numColumns > 1) || !showDivider) {
      return null;
    }
    return <Divider />;
  };
}

const styles = StyleSheet.create({
  item: {
    width: theme.viewport.width / 1.15,
    color: theme.colors.darkTint5,
  },
  itemWidth: {
    width: theme.viewport.width / 2.5,
    
  },
  redText: {
    color: theme.colors.error,
    
  },
  disable: {
    color: theme.colors.disabled,
  },
});
