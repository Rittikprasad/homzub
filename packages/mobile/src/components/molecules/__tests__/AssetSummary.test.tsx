import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AssetSummary } from '@homzhub/mobile/src/components/molecules/AssetSummary';

describe('AssetSummary', () => {
  let wrapper: ShallowWrapper;

  it('should match snapshot', () => {
    const props = {
      notification: 10,
      serviceTickets: 10,
      dues: 10,
    };
    wrapper = shallow(<AssetSummary {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match snapshot without values', () => {
    const props = {};
    wrapper = shallow(<AssetSummary {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
