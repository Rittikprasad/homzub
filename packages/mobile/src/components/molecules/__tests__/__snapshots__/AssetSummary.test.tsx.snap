// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetSummary should match snapshot 1`] = `
<LinearGradient
  angle={180}
  colors={
    Array [
      "#FFFFFF",
      "#F0F5F9",
    ]
  }
  end={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  start={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 4,
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  useAngle={true}
/>
`;

exports[`AssetSummary should match snapshot without values 1`] = `
<LinearGradient
  angle={180}
  colors={
    Array [
      "#FFFFFF",
      "#F0F5F9",
    ]
  }
  end={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  start={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 4,
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  useAngle={true}
/>
`;
